dev-lang/icon/icon-9.5.20i.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic toolchain-funcs
+
+DESCRIPTION="very high level language"
+HOMEPAGE="http://www.cs.arizona.edu/icon/"
+
+SRC_URI="https://github.com/gtownsend/icon/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="public-domain HPND"
+SLOT="0"
+KEYWORDS="~alpha ~amd64 ~hppa ~ia64 ~ppc ~ppc64 ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos"
+IUSE="X iplsrc"
+
+S="${WORKDIR}/icon-${PV}"
+
+RDEPEND="
+	X? ( x11-libs/libX11:= )"
+DEPEND="
+	${RDEPEND}
+	X? (
+		x11-libs/libXpm
+		x11-libs/libXt
+	)"
+
+src_prepare() {
+	default
+	# Add CFLAGS and LDFLAGS to Makefiles as needed
+	sed -i -e 's/CFLAGS="$(CFLAGS)"/& LDFLAGS="$(LDFLAGS)"/g' ipl/cfuncs/Makefile || die
+	sed -i -e 's/^CFLAGS = \(.*\)/CFLAGS = $(CFLAGS) \1/g' ipl/packs/loadfunc/Makefile || die
+	sed -i -e 's/CFLAGS="$(CFLAGS)"/& LDFLAGS="$(LDFLAGS)"/g' ipl/packs/loadfunc/Makefile || die
+	sed -i -e 's/-o $LIBNAME $CFLAGS/& $LDFLAGS/g' ipl/cfuncs/mklib.sh || die
+	sed -i -e 's/$LIBNAME -fPIC/$LIBNAME -fPIC $LDFLAGS/g' ipl/cfuncs/mklib.sh || die
+	sed -i -e 's/$(RLINK)/& $(LDFLAGS)/g' src/runtime/Makefile || die
+
+	# executables in src/common are only used to prep the build
+	sed -i -e 's/$(CC) $(CFLAGS)/& $(LDFLAGS)/g' src/common/Makefile || die
+	# executables in src/rtt are only used to perform the build
+	sed -i -e 's/$(CC) $(CFLAGS)/& $(LDFLAGS)/g' src/rtt/Makefile || die
+
```

Comment:
This is just.. *sigh*. Any chance, at some other time, to patch these and preferably straight in upstream?

Reply:
I've asked in the past.  The upstream maintainer has been reluctant to make any changes to the build process.  Upstream has been stabilized and has typically only been taking updates when there has been a bug found (build failures, or runtime related bugs).


As a quick side note, any new work on the language has been happening in the Unicon dialect (which is a different project)